### PR TITLE
added an option to extract more metadata from crawled websites

### PR DESCRIPTION
**Pull request by woodworker**
# Sitemap Metadata extractor

this pr makes it possible to extract more metadata from websites for later use.

my usecase:
parsing ld+json or microdata from sites and store it as structured data in the metadata field

### CODE REVIEW

File: langchain/document_loaders/sitemap.py [hunk 1]
```diff
@@ -110,7 +115,7 @@ def load(self) -> List[Document]:
         return [
             Document(
                 page_content=self.parsing_function(results[i]),
-                metadata={**{"source": els[i]["loc"]}, **els[i]},
+                metadata={**{"source": els[i]["loc"]}, **self.meta_function(els[i], results[i])},
```
Comment (dev2049) [inline langchain/document_loaders/sitemap.py:118]: can this just be `metadata=self.meta_function(els[i], results[i])`?

File: langchain/document_loaders/sitemap.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
 def _default_parsing_function(content: Any) -> str:
     return str(content.get_text())
 
+def _default_meta_function(list: dict, _content: Any) -> dict:
```
Comment (dev2049) [inline langchain/document_loaders/sitemap.py:13]: could we rename `list`? something like `element` or `meta` or `metadata`?

also do you have an example where _content is needed?

**Review by dev2049 — COMMENTED** [review]
two small comments but otherwise looks good, thanks @woodworker!

File: langchain/document_loaders/sitemap.py (hunk 1 above)
Comment (woodworker) [reply langchain/document_loaders/sitemap.py:118]: here i was not sure if i should keep the source as field that always exists or if i should move the loc to source copy to the meta function.
i can move this into the default meta function and add a comment above to remember to extract source yourself if you overwrite it

File: langchain/document_loaders/sitemap.py (hunk 2 above)
Comment (woodworker) [reply langchain/document_loaders/sitemap.py]: i renamed the parameter to meta

for an example, i added another integration test that shows what could be done - i extracted the title text in the test

**Comment by dev2049** [comment]
thanks @woodworker!

**Pull request merged**